app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingThemePickerViewHolder.kt

```diff
@@ -19,4 +24,71 @@ class OnboardingThemePickerViewHolder(private val view: View) : RecyclerView.Vie
     companion object {
         const val LAYOUT_ID = R.layout.onboarding_theme_picker
     }
+
+    init {
+        val radioLightTheme = view.theme_light_radio_button
+        val radioDarkTheme = view.theme_dark_radio_button
+        val radioFollowDeviceTheme = view.theme_automatic_radio_button
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+            radioFollowDeviceTheme?.key = R.string.pref_key_follow_device_theme
+        } else {
+            radioFollowDeviceTheme?.key = R.string.pref_key_auto_battery_theme
+        }
+
+        radioLightTheme.addToRadioGroup(radioDarkTheme)
+        radioDarkTheme.addToRadioGroup(radioLightTheme)
+
+        radioLightTheme.addToRadioGroup(radioFollowDeviceTheme)
+        radioDarkTheme.addToRadioGroup(radioFollowDeviceTheme)
+
+        radioFollowDeviceTheme.addToRadioGroup(radioDarkTheme)
+        radioFollowDeviceTheme.addToRadioGroup(radioLightTheme)
+
+        view.theme_dark_image.setOnClickListener {
+            radioDarkTheme.performClick()
+        }
+
+        view.theme_light_image.setOnClickListener {
+            radioLightTheme.performClick()
+        }
+
+        view.clickable_region_automatic.setOnClickListener {
+            radioFollowDeviceTheme.performClick()
+        }
+
+        radioLightTheme.onClickListener {
+            setNewTheme(AppCompatDelegate.MODE_NIGHT_NO)
+        }
+
+        radioDarkTheme.onClickListener {
+            setNewTheme(AppCompatDelegate.MODE_NIGHT_YES)
+        }
+
+        radioFollowDeviceTheme.onClickListener {
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+                setNewTheme(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
+            } else {
+                setNewTheme(AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY)
+            }
+        }
+
+        with(Settings.getInstance(view.context)) {
+            when {
+                this.shouldFollowDeviceTheme -> radioFollowDeviceTheme.isChecked = true
+                this.shouldUseLightTheme -> radioLightTheme.isChecked = true
+                else -> radioDarkTheme.isChecked = true
+            }
+        }
+    }
+
+    private fun setNewTheme(mode: Int) {
+        if (AppCompatDelegate.getDefaultNightMode() == mode) return
+        AppCompatDelegate.setDefaultNightMode(mode)
+        view.context?.asActivity()?.recreate()
+        view.context?.components?.core?.let {
```

Comment:
Should this be handled inside the theme manager?

Reply:
I could rename the function but it's not actually changing the Fenix theme, it's always "NormalTheme" but just toggling the system day/night setting.